### PR TITLE
docs: Adds section for state type restrictions

### LINKED ISSUE #809: Inform users about immer workings for complex objects
Easy-peasy documentation is implying that mutations can be done on all parts of the state, without specifying that they are some [restrictions for Classes, Maps and Sets](https://immerjs.github.io/immer/complex-objects/)

I lost a few hours figuring out that my mutations on a class instance weren't triggering a new rendering because it was missing `[immerable]=true`. The documentation about [modifying the state](https://easy-peasy.vercel.app/docs/tutorials/primary-api.html#modifying-the-state) could be updated to warn other users about this. 

Something like:
"/!\ While this applies to array and plain JS objects, other objects may not be immutable by default. Refer to immer documentation for more info"

Many thanks for this library and your work.

### PR DESCRIPTION
**Pull request by jmyrland**
Adds a section explaining how to support complex objects in state.

[Preview: State type restrictions](https://easy-peasy-git-fb-docscomplex-types-ctrlplusb.vercel.app/docs/tutorials/primary-api.html#state-type-restrictions)

closes #809

### CODE REVIEW

File: website/docs/docs/tutorials/primary-api.md [hunk 1]
```diff
@@ -99,6 +101,21 @@ const model = {
...
+
+If you however want to use complex types as part of your state, you need to
+consider the following:
+
+- To support `class` instances, you need to ensure that the `class` is
+  [compatible with `immer`](https://immerjs.github.io/immer/complex-objects/).
+- To use `Map` or `Set` within your state,
+  [you need to enable it](/docs/introduction/browser-support.html#supporting-map-or-set-within-your-state).
+
+⚠️ Using complex types within the state may lead to unknown side effects.
```
Comment (jmyrland) [inline website/docs/docs/tutorials/primary-api.md:117]: I'm not sure about this one - it's just a "cover your ass"-statement 😁

Keep or remove it?

Comment (no-stack-dub-sack) [reply website/docs/docs/tutorials/primary-api.md:117]: I say keep it. This covers three known cases, but there may be other things out there that haven't surfaced yet (seems like there would be limited use-cases for someone to actually do this, but what about `WeakMap` and `WeakSet` - if the polyfill doesn't cover those...). CYA never hurts.

File: website/docs/docs/tutorials/primary-api.md [hunk 2]
```diff
@@ -99,6 +101,21 @@ const model = {
 };
 ```
 
+### State type restrictions
+
+Using plain objects as state is recommended, as it has been battle tested &
+proven to be stable.
+
+If you however want to use complex types as part of your state, you need to
```
Comment (no-stack-dub-sack) [inline website/docs/docs/tutorials/primary-api.md:109]:
```suggestion
If you, however, want to use complex types as part of your state, you need to
```
Optional suggestion.

File: website/docs/docs/tutorials/primary-api.md [hunk 3]
```diff
@@ -99,6 +101,21 @@ const model = {
 };
 ```
 
+### State type restrictions
+
+Using plain objects as state is recommended, as it has been battle tested &
```
Comment (no-stack-dub-sack) [inline website/docs/docs/tutorials/primary-api.md:106]:
```suggestion
Using plain objects as state is recommended, as it has been battle tested and
```
Optional suggestion.

**Review by no-stack-dub-sack — DISMISSED** [review]
LGTM, with a couple of minor, optional suggestions.

**Comment by jmyrland** [comment]
> LGTM, with a couple of minor, optional suggestions.

Guess I need a new review after applying the changes 😁

**Review by no-stack-dub-sack — APPROVED** [review]
👍 Looks good!

**Pull request merged**